src/main/java/tak/TakServer.java

```diff
@@ -6,58 +6,82 @@
 package tak;
 
 import java.io.IOException;
+import java.net.InetSocketAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
 import java.net.SocketTimeoutException;
 import java.util.Date;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import com.sun.net.httpserver.HttpServer;
+
+import tak.httpHandlers.AddSeekHandler;
+
 /**
  *
  * @author chaitu
  */
-public class TakServer extends Thread{
+public class TakServer extends Thread {
 	public static int port;
 	public static int portws;
+	public static int portHttp;
+
+	protected static Logger logger = Logger.getLogger(TakServer.class.getName());
 
 	@Override
-	public void run () {
+	public void run() {
 		ServerSocket ssocket;
 		ServerSocket wsocket;
+		HttpServer httpServer;
+		var gameUpdateBroadcaster = new GameUpdateBroadcaster();
+		new Thread(gameUpdateBroadcaster).start();
+
 		try {
 			ssocket = new ServerSocket(port);
-			System.out.println("Server running at " + port);
+			Log("Server running at " + port);
 			ssocket.setSoTimeout(70);
 			wsocket = new ServerSocket(portws);
-			System.out.println("WebSocket Server running at " + portws);
+			Log("WebSocket Server running at " + portws);
 			wsocket.setSoTimeout(70);
-			while(true) {
-				try{
+
+			// HTTP Server example from https://stackoverflow.com/questions/3732109/simple-http-server-in-java-using-only-java-se-api
+			httpServer = HttpServer.create(new InetSocketAddress(portHttp), 0);
+			Log("HTTPServer running at " + portHttp);
+
+			// TODO: this matches `/api/v1/seeks*` but should only match exactly `/api/v1/seeks`
+			httpServer.createContext("/api/v1/seeks", new AddSeekHandler());
+			// Enables parallel processing of requests, probably not necessary
+			httpServer.setExecutor(java.util.concurrent.Executors.newCachedThreadPool());
+			httpServer.start();
+
+			while (true) {
+				try {
 					Date date = new Date();
-					if((date.getTime()&128)==128){
+					if ((date.getTime() & 128) == 128) {
```

Comment:
btw Anyone knows what this is for? Looks like `If the timestamp (in milliseconds) is a multiple of 128 then check the telnet socket, otherwise check websocket` - wat?!

Reply:
I have no idea, black magic that does something 🤷

Reply:
! I got obsessed with this question and have an answer?

- code archeology didn't help, there's a big jump from chaitu's version and nohat's version

- going back to the definition of bitwise AND... It seems to basically be like N % 128 != 0. Or:

- for every 128 milliseconds: the first time, check for telnet, the other 127 times, check for websockets

Not 100% sure about how the sockets work exactly, I don't know why we need to split it like that

Reply:
....sorry Nitzel already got there

Reply:
Definitely do keep it for now, despite being magical.

It looks like serversocket.accept blocks until the timeout, which is 70msecs, so it's more it checks for one socket for 70msecs, then depending on the "& 128" will check either the telnet or websocket server socket.

A surprise because I thought had more multi threading stuff?

Reply:
Ah the 70ms bit explains it @devp , nice find!

The accepted `Client`s are a `Thread` each, so they run concurrently.